puppet/lib/puppet/reports/puppetdb.rb

```diff
@@ -159,8 +159,13 @@ def run_duration
   # @api private
   def event_to_hash(event)
     corrective_change = defined?(event.corrective_change) ? event.corrective_change : nil
+
+    # PUP-1542 sometimes causes status to be nil; fix it up here so we
+    # can store the rest of the report
+    status = event.status || 'failure'
```

Comment:
I'm not sure where the agent team landed with this (maybe this is what we want), but it still feels like we're opening ourselves up to bad data here. The current context where this can happen is understood, but we could end up in other situations down the road where this becomes nil for other reasons.

Maybe we should add a note on the PUP ticket to remove this defaulting when it's resolved?